src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj

```diff
@@ -45,5 +52,30 @@
     <NuspecProperty Include="PackageThirdPartyNoticesFile=$(PackageThirdPartyNoticesFile)" />
   </ItemGroup>
 
-
+  <!--
+    As versioning changes, cannot compile against both the ASP.NET runtime / ref assemblies in the SDK e.g. a 5.0 SDK
+    and Microsoft.AspNetCore.Components.WebAssembly.Server (which would be 6.0 in this example). Ensure we build using
+    the targeting pack that's just been built.
+  -->
+  <Target Name="_UpdateSharedFrameworkAssemblies"
+      AfterTargets="ResolveFrameworkReferences"
+      BeforeTargets="ResolveTargetingPackAssets;_PrepareForReadyToRunCompilation;ResolveFrameworkReferencesDesignTime;_WarnAboutRedundantRef"
+      DependsOnTargets="ResolveFrameworkReferences">
```

Comment:
@rainersigwald I added `AfterTargets` because the `ResolveFrameworkReferences` has a `Returns` and thought that indicated it might get used from an outer build. But, that doesn't seem to be the case, at least in the cases I checked. In addition, this project has just a single target framework as seen above -- https://github.com/dotnet/aspnetcore/pull/24983/files#diff-eef634c0ab4b1d011d44f35afc0ed324R4
So, should we need both `AfterTargets` and `DependsOnTargets`❔

FYI the overall aim here is to make sure we fix the resolved targeting pack and framework reference metadata before anything tries to use it.

Reply:
Having it listed in both is not unreasonable.

`AfterTargets` means "whenever `ResolveFrameworkReferences` runs, run this right afterward."

`DependsOnTargets` means "if this is going to run, run `ResolveFrameworkReferences` first."

In the normal flow of things, `ResolveFrameworkReferences` runs because of other dependencies (in the common targets/the SDK). So your `AfterTargets` probably generally hooks to that, and might be sufficient on its own.

`DependsOnTargets` would make sense if the target dependency flow goes through this target--because one of its `BeforeTargets` wants to run, so this runs, and the `DependsOn` forces `ResolveFrameworkReferences` to run.